pkg/bff/auth/auth_test.go

```diff
@@ -131,6 +135,38 @@ func TestAuthenticate(t *testing.T) {
 	require.Equal(t, "invalid authorization token", rep["error"])
 }
 
+func TestAuthenticatePublicKeys(t *testing.T) {
+	// Creates a test server that serves well known jwks keys instead of the Auth0
+	// tenant - used to mock Auth0 (not a live test) but checks the happy path.
+	// NOTE: this test is fragile, e.g. if Auth0 changes its implementation.
+	srv, err := authtest.New()
+	require.NoError(t, err, "could not create authtest server")
+	defer srv.Close()
+
+	// Setup authentication middleware
+	authenticate, err := auth.Authenticate(srv.Config(), auth.WithHTTPClient(srv.Client()))
+	require.NoError(t, err, "expected valid authenticate middleware")
+
+	// Create default handler
+	success := func(c *gin.Context) {
+		c.JSON(http.StatusOK, api.Reply{Success: true})
+	}
+
+	// Create a valid token to authenticate
+	tks, err := srv.NewToken()
+	require.NoError(t, err, "could not create valid token")
+
+	// Execute request expecting success
+	c, s, w := createTestContext(http.MethodGet, "/", nil, authenticate, success)
```

Comment:
`c` is pretty standard for the [`gin.Context`](https://github.com/gin-gonic/gin#quick-start) and I don't want to confuse it with a `context.Context` which is usually `ctx`. Similarly, `w` is standard for [`http.HandlerFunc`](https://pkg.go.dev/net/http#HandlerFunc.ServeHTTP). 

`s` is not standard for the router so I could change that to `srv` or `mux` if it would make it easier to read and add a comment to make clearer. Do you have other suggestions for other names? We can create a new PR to make the code more self-documenting if you do.